### ISSUE TITLE
Elasticsearch 8.15 fails bootstrap check without additional dependencies

### ISSUE THREAD

**Issue by mousumis** [elastic/elasticsearch#115104]
### Elasticsearch Version

8.15.2

### Java Version

17.0.12

### OS Version

Amazon Linux 2 kernel 5.15.167-112.165.amzn2.x86_64 / Alpine Linux 3.2.0

### Problem Description

The 8.15 Elasticsearch upgrade seems to have changed the way bootstrap checks are done which results in startup failing without certain new dependencies which were not previously mandated ([this commit](https://github.com/elastic/elasticsearch/commit/bc499e7c830d55c94042f141e4c1501348f397aa) may be part of the issue). This is not documented in the Release Notes or Breaking Changes for 8.15 or 8.15.2 in particular. We have tried in docker containers with 2 types of OS's and the bootstrap check fails on both due to missing dependencies

- AL2: libzstd not found
- Alpine Linux: libsystemd not found

For AL2, installing libzstd did allow startup to continue. For Alpine Linux, there is no good solution since the libsystemd is not available for install at all. This has been reported in the forums where there has been no resolution - https://discuss.elastic.co/t/bootstrapcheck-using-a-non-available-library/365826

Moreover, this seems to be a bug or unintended side effect of the changes made for 8.15 and is blocking our upgrade since we cannot account for additional libraries being installed for a minor version bump. This should be addressed so that installs on 8.14 can continue using the same infrastructure.

A post was already made in Elastic forums which has not had any responses - https://discuss.elastic.co/t/bootstrap-check-failure-in-8-15-2/368660

### Steps to Reproduce

- Our setup includes a docker container based off Amazon Linux 2
- We install elasticsearch distribution from https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-8.15.2-windows-x86_64.zip and make some modifications to deploy it across multiple platforms
- We use our own Java 17 environment, not the one shipped with the distribution
- We attempt to start using the bin/elasticsearch binary but the bootstrap checks fail before we get to any further steps due to the libzstd dependency. This does not halt startup in 8.14

- In addition, we run automated tests on Alpine linux where startup fails due to libsystemd not being found and there does not seem to be a good way to work around this due to libsystemd not being supported on Alpine.

### Logs (if relevant)

**AL2**:
```
java.lang.UnsatisfiedLinkError: Unable to load library 'zstd':
libzstd.so: cannot open shared object file: No such file or directory
libzstd.so: cannot open shared object file: No such file or directory
Native library (linux-x86-64/libzstd.so) not found in resource path ()
........
........
[ERROR][org.elasticsearch.bootstrap.Elasticsearch] [server-deployment] node validation exception
[2] bootstrap checks failed. You must address the points described in the following [2] lines before starting Elasticsearch. For more information see [https://www.elastic.co/guide/en/elasticsearch/reference/8.15/bootstrap-checks.html]
bootstrap check failure [1] of [2]: max size virtual memory [-1] for user [<redacted>] is too low, increase to [unlimited]; for more information see [https://www.elastic.co/guide/en/elasticsearch/reference/8.15/max-size-virtual-memory-check.html]
bootstrap check failure [2] of [2]: max file size [-1] for user [<redacted>] is too low, increase to [unlimited]; for more information see [https://www.elastic.co/guide/en/elasticsearch/reference/8.15/_max_file_size_check.html]
```

**Alpine**:
```
java.lang.UnsatisfiedLinkError: Unable to load library 'libsystemd.so.0':
Error loading shared library libsystemd.so.0: No such file or directory
Error loading shared library libsystemd.so.0: No such file or directory
Native library (linux-x86-64/libsystemd.so.0) not found in resource path ()
........
........
[ERROR][org.elasticsearch.bootstrap.Elasticsearch] [localhost] node validation exception
[2] bootstrap checks failed. You must address the points described in the following [2] lines before starting Elasticsearch. For more information see [https://www.elastic.co/guide/en/elasticsearch/reference/8.15/bootstrap-checks.html]
bootstrap check failure [1] of [2]: max size virtual memory [-1] for user [<redacted>] is too low, increase to [unlimited]; for more information see [https://www.elastic.co/guide/en/elasticsearch/reference/8.15/max-size-virtual-memory-check.html]
bootstrap check failure [2] of [2]: max file size [-1] for user [<redacted>] is too low, increase to [unlimited]; for more information see [https://www.elastic.co/guide/en/elasticsearch/reference/8.15/_max_file_size_check.html]
```

**Comment by thecoop** [comment]
There's a few things going on here

- You are attempting to use the windows distribution to deploy onto Amazon Linux 2? Needless to say, this is not supported, and I'm not surprised you are encountering linker issues.
- The 'max size -1' issues look like a duplicate of [this one](https://github.com/elastic/elasticsearch/issues/113705) - you can find more information there, and if you can provide us a supported reproduction case that would help us to find a solution.
- Alpine Linux is not a supported OS. You can find our OS support matrix [here](https://www.elastic.co/support/matrix#matrix_os). We expect to run on OSs using systemd, and do not support other OSs (you can find a similar issue [here](https://github.com/elastic/elasticsearch/issues/113666)). Given that, we do have a change that makes it possible to start Elasticsearch without systemd in 8.16, and I recommend you try 8.16 when it is released.

**Comment by mousumis** [comment]
@thecoop Thank you for your feedback. I understand that our deployment is customized due to other configurations we need to support. My main concern is that a setup that worked for 8.14 no longer works for 8.15 and we could not find documented about it. Based on you what you said it sounds like 8.16 may address them so we can plan on taking the update.

Is there an estimate for when it will be available?

**Comment by ldematte** [comment]
I think that @thecoop addressed all the points; if you still have problems after upgrading to 8.16, especially with https://github.com/elastic/elasticsearch/issues/113705, please do comment on that issue with reproduction steps or logs.
I'll make a note to discuss with the team about documenting the zstd requirement, it seems we do not state that clearly in our Support Matrix.
For Alpine Linux, we cannot guarantee it works, as it's not a supported OS (we don't test on it, so we have no way of knowing/support it). The change in 8.16 _should_ work, as it removes the dependency on libsystemd, but we still expect to run on OSs that use systemd, so there is no guarantee.

**Comment by rjernst** [comment]
> I'll make a note to discuss with the team about documenting the zstd requirement

TBC, there is _no_ zstd requirement. We ship zstd with Elasticsearch. The bug here was in how we find libraries, both libsystemd (which we no longer need as of 8.16) and libzstd (which we ship with and can now find).

**Comment by ldematte** [comment]
Thank you for correcting my statement Ryan
